### PR TITLE
Analyser 2.0: back to the roots

### PR DESCRIPTION
**Pull request by DavidBadura**
## Gadgets

* [x] phpcs
* [x] phpmd
* [x] pdepend
* [x] twig-lint
* [x] security-checker
* [x] ~comment blacklist~ replace with: https://github.com/squizlabs/PHP_CodeSniffer/tree/master/CodeSniffer/Standards/Generic/Sniffs/Commenting
* [x] ~function-blacklist~ replace with:
https://github.com/squizlabs/PHP_CodeSniffer/blob/master/CodeSniffer/Standards/Generic/Sniffs/PHP/ForbiddenFunctionsSniff.php
* [x] ~doc-block~ todo: create a separate lib or replace with an existing lib

## Solved / Closed Issues

fix: #66, #53, #54, #71, #40, #37 

## Allows to solve the following problems

* #59
* #75
* #6

### CODE REVIEW

**Comment by DavidBadura** [comment]
I think secuirty check should be an another projekt like [VersionEye](https://www.versioneye.com/). This has nothing to do with simpspector's focus (code-style, improve pull requests).

**Comment by tolry** [comment]
@DavidBadura isn't security warnings also some kind of static code analysis

general question: this ist still WIP, right?

**Comment by DavidBadura** [comment]
yes, it is still WIP.

1.) If you have an old project, you create a new PR and get an security issue, then it's not your PR's fault. And sometimes you can't fix it in this PR. You need to create a new PR. 

2.) You want to check security issues in interval (every day). But simpspector only runs on a new commit.

Security-Check in simpsector is a nice addon, but i see it in an another project.

**Comment by DavidBadura** [comment]
ping @tolry @smoench @larsborn @dragonito

File: config/parameters.dist.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+parameters:
```
Comment (tolry) [inline config/parameters.dist.yml]: Make it a bit more obvious, that dist.yml is used as default values and not only as a template for parameters.yml? e.g. small comment above something like?
``` yml
# default parameter values, can be ovewritten in parameters.yml
```

File: config/default_simpspector.yml [hunk 2]
```diff
@@ -1,21 +0,0 @@
-phpcs:
```
Comment (tolry) [inline config/default_simpspector.yml]: No more default_simpspector.yml at all? Where can I find a template, which gadgets can be activated? Preferably with a comment per gadget with a link to how this gadget is configured (config reference)

File: src/Gadget/AbstractGadget.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * @author David Badura <d.a.badura@gmail.com>
+ */
+abstract class AbstractGadget implements GadgetInterface
+{
+    /**
+     * @param string $path
+     * @param array $args
+     * @param AbstractLogger $logger
```
Comment (tolry) [inline src/Gadget/AbstractGadget.php]: We already discussed this naming issue here, didn't we? For another day I guess ;-)

Comment (DavidBadura) [inline src/Gadget/AbstractGadget.php]: see https://github.com/simpspector/analyser/issues/57

Btw: https://github.com/simpspector/analyser/milestone/3

File: .travis.yml [hunk 4]
```diff
@@ -21,7 +19,10 @@ env:
...
     - travis_retry composer self-update
     - travis_retry composer update $PREFER_LOWEST
+    - curl -OL https://squizlabs.github.io/PHP_CodeSniffer/phpcs.phar
+    - chmod +x phpcs.phar
+    - sed -i -e 's/\"phpcs\"/\"\.\/phpcs\.phar\"/g' config/parameters.dist.yml
 
 script:
     - bin/phpunit
-    - bin/simpspector analyse ./ -v -f detail
+    - bin/simpspector analyse ./ -v -f detail
```
Comment (larsborn) [inline .travis.yml]: WHHHHYYYY?

File: src/Console/Application.php [hunk 5]
```diff
@@ -34,18 +34,15 @@ public function __construct(
         parent::__construct('SimpSpector', 'dev');
 
         $this->add(new AnalyseCommand($analyser, $formatter));
-        $this->add(new ReferenceCommand($repository));
         $this->add(new DiffCommand($importer, new Calculator(), $analyser->getExecutor(), $analyser->getLoader()));
     }
 
     /**
      * @param string|null $bin
```
Comment (DavidBadura) [inline src/Console/Application.php]: anpassen

**Pull request merged**